posts/2024-10-17-Rust-1.82.0.md

```diff
@@ -0,0 +1,320 @@
+<b><span style="color:#0A0">crates.io:</span></b> https://crates.io/crates/cc/1.1.23
+<b><span style="color:#0A0">features:</span></b>
+  jobserver = []
+  parallel  = [dep:libc, dep:jobserver]
+<b><span style="color:#0AA">note</span></b><b>:</b> to see how you depend on cc, run `<b><span style="color:#0AA">cargo tree --invert --package cc@1.1.23</span></b>`
+</pre>
+
+By default `cargo info` describes the package version in the local `Cargo.lock`, if any, but as you can see it will indicate when there's a newer version too, and `cargo info cc@1.1.30` would report on that. The feature list also indicates a `+` next to any that are enabled.
+
+### Apple target promotions
+
+#### macOS on ARM64 is now Tier 1
+
+The Rust target `aarch64-apple-darwin` for macOS on ARM64 (M1-family or later Apple Silicon CPUs) is now a tier 1 target, indicating our highest guarantee of working properly. As the [platform support](https://doc.rust-lang.org/stable/rustc/platform-support.html) page describes, every change in the Rust repository must pass full tests on every tier 1 target before it can be merged. This ARM64 target was added to tier 2 back in Rust 1.49, making it available in `rustup`, and this new milestone raises it on par with ARM64 Linux and the X86 macOS, Linux, and Windows targets.
+
+#### Mac Catalyst targets are now Tier 2
+
+[Mac Catalyst](https://developer.apple.com/mac-catalyst/) is a technology by Apple that allows running iOS applications natively on the Mac. This is especially useful when testing iOS-specific code, as `cargo test --target=aarch64-apple-ios-macabi --target=x86_64-apple-ios-macabi` mostly just works (in contrast to the usual iOS targets, which need to be bundled using external tooling before they can be run on a native device or in the simulator).
+
+[The targets](https://doc.rust-lang.org/nightly/rustc/platform-support/apple-ios-macabi.html) are now tier 2, and can be downloaded with `rustup target add aarch64-apple-ios-macabi x86_64-apple-ios-macabi`, so now is an excellent time to update your CI pipeline to test that your code also runs in iOS-like environments.
+
+### Minimal exhaustive patterns
+
+Empty patterns can now be omitted in common cases:
+
+```rust
+use std::convert::Infallible;
+pub fn safe_unwrap<T>(x: Result<T, Infallible>) -> T {
+    let Ok(x) = x; // the `Err` case does not need to appear
+    x
+}
+```
+
+This works with empty types such as a variant-less `enum Void {}`, or structs and enums with a visible empty field and no `#[non_exhaustive]` attribute. It will also be particularly useful in combination with the never type `!`, although that type is still unstable at this time.
+
+This feature is "minimal" because there are still some exclusions at this time. For reasons related to uninitialized values and unsafe code, omitting patterns is not allowed if the empty type is accessed through a reference, pointer, or union field:
```

Comment:
This suggestion goes with my earlier rewording of the section heading which removed “minimal”, and doesn't make sense otherwise:

```suggestion
There are still some cases where empty patterns must still be written. For reasons related to uninitialized values and unsafe code, omitting patterns is not allowed if the empty type is accessed through a reference, pointer, or union field:
```